File: segmentController/src/main/java/com/clumob/segment/controller/SegmentControllerImpl.java
package com.clumob.segment.controller;

public class SegmentControllerImpl<T> implements SegmentController<T> {

    private final Storable args;
    private final T viewData;

    public SegmentControllerImpl(Storable args, T viewData) {
        this.args = args;
        this.viewData = viewData;
    }

    @Override
    public T getViewData() {
        return viewData;
    }

    @Override
    public void onCreate() {

    }

    @Override
    public void restoreState(Storable restorableState) {

    }

    @Override
    public void onStart() {

    }

    @Override
    public void onResume() {

    }

    @Override
    public void onPause() {

    }

    @Override
    public void onStop() {

    }

    @Override
    public void onDestroy() {

    }
}

File: segmentController/src/main/java/com/clumob/segment/controller/SegmentControllerImpl.kt
package com.clumob.segment.controller

open class SegmentControllerImpl<T>(private val args: Storable?, override val viewData: T) : SegmentController<T> {

    override fun onCreate() {}
    override fun restoreState(restorableState: Storable?) {}
    override fun onStart() {}
    override fun onResume() {}
    override fun onPause() {}
    override fun onStop() {}
    override fun onDestroy() {}

}